packages/features/eventtypes/components/tabs/assignment/EventTeamAssignmentTab.tsx

```diff
@@ -252,24 +254,33 @@ const RoundRobinHosts = ({
         <p className="text-subtle max-w-full break-words text-sm leading-tight">{t("round_robin_helper")}</p>
       </div>
       <div className="border-subtle rounded-b-md border border-t-0 px-6 pt-4">
-        {!assignAllTeamMembers && !assignRRMembersUsingSegment && (
-          <Controller<FormValues>
-            name="isRRWeightsEnabled"
-            render={({ field: { value, onChange } }) => (
-              <SettingsToggle
-                title={t("enable_weights")}
-                description={weightDescription}
-                checked={value}
-                onCheckedChange={(active) => {
-                  onChange(active);
+        {!assignRRMembersUsingSegment && (
+          <>
+            <Controller<FormValues>
+              name="isRRWeightsEnabled"
+              render={({ field: { value: isRRWeightsEnabledValue, onChange } }) => (
+                <SettingsToggle
+                  title={t("enable_weights")}
+                  description={weightDescription}
+                  checked={isRRWeightsEnabledValue}
+                  onCheckedChange={(active) => {
+                    onChange(active);
 
-                  const rrHosts = getValues("hosts").filter((host) => !host.isFixed);
-                  const sortedRRHosts = rrHosts.sort((a, b) => sortHosts(a, b, active));
-                  setValue("hosts", sortedRRHosts);
-                }}
-              />
-            )}
-          />
+                    const rrHosts = getValues("hosts").filter((host) => !host.isFixed);
+                    const sortedRRHosts = rrHosts.sort((a, b) => sortHosts(a, b, active));
+                    setValue("hosts", sortedRRHosts);
+                  }}>
+                  {!assignRRMembersUsingSegment ? (
+                    <EditWeightsForAllTeamMembers
+                      teamMembers={teamMembers}
```

Comment:
Shouldn't we only show assigned hosts here? Now we always show all host's even when 'Add all team members, including future members' is  not enabled